components/input/input-otp.component.ts

```diff
@@ -0,0 +1,229 @@
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    @for (item of otpArray.controls | keyvalue; track $index) {
+      <input
+        nz-input
+        class="ant-otp-input"
+        type="text"
+        maxlength="1"
+        size="1"
+        [nzSize]="nzSize"
+        [formControl]="$any(otpArray.at($index))"
+        [nzStatus]="nzStatus"
+        (input)="onInput($index, $event)"
+        (focus)="onFocus($event)"
+        (keydown)="onKeyDown($index, $event)"
+        (paste)="onPaste($index, $event)"
+        #otpInput
+      />
+    }
+  `,
+  host: {
+    class: 'ant-otp'
+  },
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => NzInputOtpComponent),
+      multi: true
+    },
+    NzDestroyService
+  ],
+  imports: [FormsModule, NgForOf, NzInputDirective, ReactiveFormsModule, KeyValuePipe],
+  standalone: true
+})
+export class NzInputOtpComponent implements ControlValueAccessor, OnChanges {
+  @ViewChildren('otpInput') otpInputs!: QueryList<ElementRef>;
+
+  @Input({ transform: numberAttribute }) nzLength: number = 6;
+  @Input() nzSize: NzSizeLDSType = 'default';
+  @Input({ transform: booleanAttribute }) disabled = false;
+  @Input() nzStatus: NzStatus = '';
+  @Input() nzFormatter: (value: string) => string = value => value;
+  @Input() nzMask: string | null = null;
+
+  protected otpArray!: FormArray;
+  private internalValue: string[] = [];
+  private onChangeCallback?: (_: NzSafeAny) => void;
+  onTouched: OnTouchedType = () => {};
+
+  constructor(
+    private readonly formBuilder: FormBuilder,
+    private readonly nzDestroyService: NzDestroyService
+  ) {
+    this.createFormArray();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['nzLength']?.currentValue && changes['nzLength'].currentValue !== changes['nzLength'].previousValue) {
+      this.createFormArray();
+    }
+
+    if (changes['disabled']) {
+      this.setDisabledState(this.disabled);
+    }
+  }
+
+  onInput(index: number, event: Event): void {
+    const inputElement = event.target as HTMLInputElement;
+    const nextInput = this.otpInputs.toArray()[index + 1];
+
+    if (inputElement.value && nextInput) {
+      nextInput.nativeElement.focus();
+    } else if (!nextInput) {
+      this.selectInputBox(index);
+    }
+  }
+
+  onFocus(event: FocusEvent): void {
+    const inputElement = event.target as HTMLInputElement;
+    inputElement.select();
+  }
+
+  onKeyDown(index: number, event: KeyboardEvent): void {
+    const previousInput = this.otpInputs.toArray()[index - 1];
+
+    if (event.key === 'Backspace') {
+      event.preventDefault();
+
+      this.internalValue[index] = '';
+      this.otpArray.at(index).setValue('', { emitEvent: false });
+
+      if (previousInput) {
+        this.selectInputBox(index - 1);
+      }
+
+      this.emitValue();
+    }
+  }
+
+  writeValue(value: string): void {
+    if (!value) return;
```

Comment:
We should handle null values ​​so that we can just clear the input box, WDYT?

Reply:
Yeah i hadn't taken it into consideration 👍